### ISSUE TITLE
Load Mbtiles file for tiles to use google map sdk function in offline mode

### ISSUE THREAD

**Issue by syle20091** [BruTile/BruTile#90]
Hey,
I use google map SDK to display clusters and polyline in my application. Now, we would like to support offline mode. I did some research and it seems that google map functions can be used if we provided custom tiles. I got local Mbtile files from openmaptiles.com.

So I am wondering:
Can I use this lib to retrieve certain tiles from Mbtiles file? If yes, Is there a sample or could you point me a direction?
Thanks in advance

**Comment by pauldendulk** [comment]
Yes, that is possible. There is the mbtiles demo where you need to load an mbtiles file through the menu. You could also look at the mapsui source. There is a MbTilesSample.cs in there.

**Comment by pauldendulk** [comment]
https://github.com/Mapsui/Mapsui/blob/master/Samples/Mapsui.Samples.Common/Maps/MbTilesSample.cs

**Comment by pauldendulk** [comment]
https://github.com/Mapsui/Mapsui/blob/master/Mapsui/Layers/TileLayer.cs

**Comment by syle20091** [comment]
Hi Paul,
Thanks so much for the fast response!
In order to use the Google SDK in offline, I need to provide X, Y, Zoom level and image for the tiles (see pic). Regard to the link you provided, I saw it use MbTilesTileSource to load local mbtiles file and looks like these variables should be in MbTilesTileSource.Schema cause I found this method "mbTilesTileSource.Schema.GetOriginX()". 
<img width="412" alt="screen shot 2018-05-05 at 8 43 56 pm" src="https://user-images.githubusercontent.com/38998711/39662584-4cd74366-50a7-11e8-8abb-bff70ca30d34.png">

I looked everywhere and having trouble to find these variables especially the image. Could you give me some advice?

**Comment by pauldendulk** [comment]
You need to use an MbTilesTileSource. It has a GetTile method. 

```byte[] GetTile(TileInfo tileInfo);```

The TileIndex in the TileInfo has a Col, Row and Level which corresponds to the x, y and zoom. The byte array returned is the raw image data. You will need to create a UIImage from it. Perhaps convert it to stream as an intermediary step.
